gcloud-java-dns/src/main/java/com/google/gcloud/dns/ChangeRequestInfo.java

```diff
@@ -0,0 +1,358 @@
+    this.id = builder.id;
+    this.startTimeMillis = builder.startTimeMillis;
+    this.status = builder.status;
+  }
+
+  /**
+   * Returns an empty builder for the {@code ChangeRequestInfo} class.
+   */
+  public static Builder builder() {
+    return new BuilderImpl();
+  }
+
+  /**
+   * Creates a builder populated with values of this {@code ChangeRequestInfo}.
+   */
+  public Builder toBuilder() {
+    return new BuilderImpl(this);
+  }
+
+  /**
+   * Returns the list of {@link RecordSet}s to be added to the zone upon submitting this change
+   * request.
+   */
+  public List<RecordSet> additions() {
+    return additions;
+  }
+
+  /**
+   * Returns the list of {@link RecordSet}s to be deleted from the zone upon submitting this change
+   * request.
+   */
+  public List<RecordSet> deletions() {
+    return deletions;
+  }
+
+  /**
+   * Returns the id assigned to this {@code ChangeRequest} by the server.
+   */
+  public String id() {
+    return id;
+  }
+
+  /**
+   * Returns the time when this {@code ChangeRequest} was started by the server.
+   */
+  public Long startTimeMillis() {
+    return startTimeMillis;
+  }
+
+  /**
+   * Returns the status of this {@code ChangeRequest}.
+   */
+  public ChangeRequestInfo.Status status() {
+    return status;
+  }
+
+  com.google.api.services.dns.model.Change toPb() {
+    com.google.api.services.dns.model.Change pb =
+        new com.google.api.services.dns.model.Change();
+    // set id
+    if (id() != null) {
+      pb.setId(id());
+    }
+    // set timestamp
+    if (startTimeMillis() != null) {
+      pb.setStartTime(ISODateTimeFormat.dateTime().withZoneUTC().print(startTimeMillis()));
+    }
+    // set status
+    if (status() != null) {
+      pb.setStatus(status().name().toLowerCase());
+    }
+    // set a list of additions
+    pb.setAdditions(Lists.transform(additions(), RecordSet.TO_PB_FUNCTION));
+    // set a list of deletions
+    pb.setDeletions(Lists.transform(deletions(), RecordSet.TO_PB_FUNCTION));
+    return pb;
+  }
+
+  static ChangeRequestInfo fromPb(com.google.api.services.dns.model.Change pb) {
+    Builder builder = builder();
+    if (pb.getId() != null) {
+      builder.id(pb.getId());
+    }
+    if (pb.getStartTime() != null) {
+      builder.startTimeMillis(DateTime.parse(pb.getStartTime()).getMillis());
+    }
+    if (pb.getStatus() != null) {
+      // we are assuming that status indicated in pb is a lower case version of the enum name
+      builder.status(ChangeRequest.Status.valueOf(pb.getStatus().toUpperCase()));
+    }
+    if (pb.getDeletions() != null) {
+      builder.deletions(Lists.transform(pb.getDeletions(), RecordSet.FROM_PB_FUNCTION));
+    }
+    if (pb.getAdditions() != null) {
+      builder.additions(Lists.transform(pb.getAdditions(), RecordSet.FROM_PB_FUNCTION));
+    }
+    return builder.build();
+  }
+
+  @Override
+  public boolean equals(Object other) {
+    return (other instanceof ChangeRequestInfo)
```

Comment:
Use the getClass() comparison pattern (here and in ChangeRequest) to avoid the case of a == b and b != a.